tests/ZfrOAuth2Test/Server/Grant/AuthorizationGrantTest.php

```diff
@@ -244,7 +255,7 @@ public function testCanCreateTokenResponse($hasRefreshGrant)
 
         $response = $this->grant->createTokenResponse($request, new Client(), $owner);
 
-        $body = json_decode($response->getContent(), true);
+        $body = json_decode($response->getBody(), true);
```

Comment:
Whether or not it is intuitive, it *is* in the docblock: "Returns the remaining contents in the stream."

If you want the entire stream at once, casting it to a string is recommended, as it has the behavior you are expecting (`__toString()` is documented as "Reads all data from the stream into a string, from the beginning to end", and must attempt to seek to the beginning of the stream). 

As a working example:

```php
use Zend\Diactoros\Response;
use Zend\Diactoros\Stream;

$stream = new Stream('php://temp', 'wb+');
$stream->write(json_encode(['foo' => 'bar']));

$response = new Response($stream, 200);

$body = json_decode((string) $response->getBody(), true);
var_export($body);
```

This worked perfectly for me just now, which makes me wonder if perhaps you made your stream read-only? Note the second argument I passed to the `Stream` constructor, `wb+` — this is the mode to use when creating the stream, and ensures it's both writeable and readable. From what I can see, I have a bug in Diactoros currently; it *should* raise an exception on `write()` if the stream is not writeable, but it currently does not, which could be hiding this behavior.